### PR TITLE
Add chunk.EncodeTo to allow to pass your own buffer.

### PR DESCRIPTION
**Pull request by cyriltovena**
Useful if you want to re-use memory after flushing data.

Signed-off-by: Cyril Tovena <cyril.tovena@gmail.com>

### CODE REVIEW

File: pkg/chunk/chunk.go [hunk 1]
```diff
@@ -184,7 +184,14 @@ var writerPool = sync.Pool{
 // Encode writes the chunk into a buffer, and calculates the checksum.
 func (c *Chunk) Encode() error {
 	var buf bytes.Buffer
+	return c.EncodeTo(&buf)
```
Comment (pracucci) [inline pkg/chunk/chunk.go:187]: Can we just do `return c.EncodeTo(nil)`?

**Review by pracucci — APPROVED** [review]
LGTM (with a nit)

File: pkg/chunk/chunk.go (hunk 1 above)
Comment (cyriltovena) [reply pkg/chunk/chunk.go:187]: yup definitively.

**Review by jtlisi — APPROVED** [review]
LGTM

**Pull request merged**